### ISSUE TITLE
[A11y] Collection Landing - Link purpose

### ISSUE THREAD

**Issue by jechols** [OregonDigital/OD2#1821]
### Operating System

Windows

### Browser

Edge

### Affected page

Collection Show

### Admin user?

Yes

### Issue Description

Many links fail the "link purpose" test:

- "View All" on the top is iffy, though it isn't a hard fail due to the nearby context
- "View All" on the bottom seems to have no context, and is a fail
- The collection's featured images are links to items, but have no accessible text on them.  They get an accessible value on focus and hover, but until then any use of "list links" tools will give you items like "df66vb95p locale=en" in its output, so this is still a fail.
- "Oregon State University" links to a results page for everything with that facet value, but the accessible name is just "Oregon State University", which doesn't describe its purpose.

**Comment by jechols** [comment]
I'm going to split off the bit here that needs discussion so this is a simpler-to-address ticket....

**Comment by jechols** [comment]
I think this may have been put into QA on accident. Nothing appears to have been addressed. Or maybe there's a pending deploy? Either way, pushing back to ready.

**Comment by straleyb** [comment]
Fixed by #1935 

@jechols The only part of this that wasnt fixed was 
`Oregon State University" links to a results page for everything with that facet value, but the accessible name is just "Oregon State University", which doesn't describe its purpose.`

This is because this link exists in the metadata section. Some works are going to have massive amounts of metadata, so instead of just looking at the institution, we should work on fixing the accessibility for the entire metadata section. This would be much better in a second PR that focuses on that.

The linked PR fixes the buttons and the alt text on the featured images.
